client/tests/simulator_tests.rs

```diff
@@ -83,6 +85,66 @@ fn test_get_capability_manufacturer_id() {
     assert_eq!(received, expected);
 }
 
+#[test]
+fn test_get_random_max_size() {
+    let (_sim_lifeline, mut tpm) = get_started_tpm();
+
+    let command = GetRandomCmd {
+        bytes_requested: 4096,
+    };
+
+    // We allow panic in test cases.
+    let resp = run_command(&command, &mut tpm).expect("Failed running command.");
+
+    // Lets pull out the actual data as a slice for convenience
+    let random_slice = &resp.random_bytes.buffer[..resp.random_bytes.size as usize];
+
+    // We should have received less than what the client implementation supports.
+    assert!(random_slice.len() <= TpmtHa::UNION_SIZE);
+
+    // We should have received minimum size of SHA256 (minimum mandatory digest size).
+    assert!(random_slice.len() >= TPM2_SHA256_DIGEST_SIZE as usize);
+
+    // Print for test inspection.
+    trace!("Got random sequence: {:?}", random_slice);
+
+    // Duplicate value trap, to catch any same value sequences.
+    {
+        let mut same_twice: u16 = 0;
+        for i in 1..random_slice.len() as usize {
+            if random_slice[i - 1] == random_slice[i] {
+                same_twice += 1;
+            }
+        }
+
+        trace!(
+            "After {} values, we got duplicate value {same_twice} times.",
+            random_slice.len()
+        );
+        assert!(same_twice <= random_slice.len() as u16 / 2); // Same value repeated in half of the values unlikely with > 32 values.
+    }
+}
+
+#[test]
+fn test_get_random_small_sizes() {
+    let (_sim_lifeline, mut tpm) = get_started_tpm();
+
+    for i in 0..4 as usize {
```

Comment:
Why up to `4`? Just to test the lower-level functionality from `u32` vs `u64` that might be treated specially (ala @oddcoder's ongoing PR)?

In that case maybe worth going up to size of `u64` or `u128` to set some clear obvious boundary defined by a type.

Reply:
I have changed to test just 0 and 1. Zero requests seems to be allowed for some reason. And 1 is the minimal case.
The number of requested bytes is (by TPM spec and our request struct definition) limited to u16. So I modified the max test to handle 2 large requests. (0xFFF and 0xFFFF).